pkg/controller/registry/resolver/resolver.go

```diff
@@ -215,24 +220,24 @@ func (resolver *MultiSourceResolver) resolveCSV(sources map[registry.SourceKey]r
 		}
 
 		// registry.SourceKey for the source containing the CSV
-		csvSrcKey := firstSrcKey
+		csvSourceKey := preferredSourceKey
```

Comment:
`var csvSourceKey SourceKey` 

(easier to catch bugs where this isn't being set properly if it's not initialized to a valid value)